### DISCUSSION TITLE
Mengganti label di atas plot

### DISCUSSION THREAD

**Discussion by ToniAS21** [teamalgoritma/community#221] [Q&A] [answered]
Bagaimana cara mengganti nilai diatas visualisasi nya dari
![Screenshot (1697)](https://user-images.githubusercontent.com/83503249/180391975-7bfb5e62-9b94-4eff-94c6-6cf1a20fc13c.png)

dari `gross_income` menjadi "Gross Income", `jumlah_transaksi` menjadi "Jumlah Transaksi", dan `quantity` menjadi "Quantity" 

Berikut potongan kode :
```
# visualisasi
ggplot(data = city_pivot, aes(x = city, y = value)) +
  geom_col(aes(fill = city), position = "dodge") +
  facet_wrap(~name, scales = "free_y") +
  labs(title = "Performa Tiga Kota Berdasarkan Pendapatan Kotor, Jumlah Transaksi dan Quantity",
       subtitle = "Dari Januari 2019 sampai Maret 2019",
       x = NULL,
       y = "Nilai") +
  theme(axis.text.x = element_blank())
```
Sekian terima kasih

**Comment by idoali** [comment, marked as answer]
Bapak bisa mencoba dengan mengganti nama-nama tersebut dengan menggunakan `switch()` dan `sapply()` sebelum menggunakan `ggplot`. Kurang lebih akan menjadi seperti ini : 

```{r}
city_pivot <- pivot_longer(data = data_agg, 
                          cols = c("quantity", "gross_income", "jumlah_transaksi"), 
                          names_to="types")
city_pivot$types <- sapply(x=as.character(city_pivot$types), FUN=switch, 
                          "gross_income" = "Gross Income", 
                          "jumlah_transaksi" = "Jumlah Transaksi", 
                          "quantity" = "Quantity") 
```

Data yang akan digunakan untuk `ggplot` adalah data `city_pivot`, jenis data yang sama seperti yang bapak gunakan di code bapak. Tetapi, sebelum di visualisasikan, data tersebut diubah sedikit terlebih dulu. 

Semoga jawabannya bisa membantu pak.
